lib/openstudio-standards/standards/ashrae_90_1_prm/ashrae_90_1_prm_2019/ashrae_90_1_prm_2019_WaterHeaterMixed.rb

```diff
@@ -0,0 +1,126 @@
+    # Get the volumn of the water heater
+    volumn_m3 = water_heater_mixed.tankVolume
+    if volumn_m3.empty?
+      OpenStudio.logFree(OpenStudio::Warn, 'openstudio.standards.WaterHeaterMixed', "For #{water_heater_mixed.name}, cannot find volumn, standard will not be applied.")
+      return false
+    else
+      volumn_m3 = volumn_m3.get / comp_qty
+    end
+    volumn_gal = OpenStudio.convert(volumn_m3, 'm^3', 'gal').get
+
+    # Get the fuel type data
+    heater_prop = model_find_object(standards_data['prm_swh_bldg_type'], {'swh_building_type' => building_type_swh})
+    new_fuel_data = heater_prop['baseline_heating_method']
+    # There are only two water heater fuel type in the prm database:
+    # ("Gas Storage" and "Electric Resistance Storage")
+    # Change the prm fuel type to openstudio fuel type
+    if new_fuel_data == "Gas Storage"
+      new_fuel = "NaturalGas"
+    else
+      new_fuel = "Electricity"
+    end
+
+    # Calculate the water heater properties
+    # Most electricity or natural gas "Storage" water heater use uef to
+    # calculate the efficiency and skin loss coefficient.
+    # Only two cases do not use uef, which is Electricity, capacity > 40944.01
+    # and NaturalGas, capacity > 105000.01.
+    # The efficiency and skin loss coefficient of these two cases
+    # are calculated separately.
+    # The efficiency and skin loss coefficient ua is based on 90.1-2019 Table 7.8 and
+    # PNNL-23269 "Enhancements to ASHRAE Standard 90.1 Prototype Building Models" A 1.2
+    if (new_fuel == "Electricity") and (capacity_btu_per_hr > 40944.01)
+      water_heater_eff = 1
+      ua_btu_per_hr_per_f = (0.3 + 27.0/volumn_gal)/70
+    elsif (new_fuel == "NaturalGas") and (capacity_btu_per_hr > 105000.01)
+      ua_btu_per_hr_per_f = (capacity_btu_per_hr/800 + 110 * Math.sqrt(volumn_gal))*0.8/70
+      water_heater_eff = (ua_btu_per_hr_per_f * 70 + capacity_btu_per_hr * 0.8)/capacity_btu_per_hr
+    else
+      search_criteria = {}
+      search_criteria['template'] = template
+      search_criteria['fuel_type'] = new_fuel
+      if new_fuel == "Electricity"
+        search_criteria['product_class'] = "Water Heaters"
+      else
+        search_criteria['product_class'] = "Storage Water Heater"
+      end
+
+      # Todo: Use 'medium' as draw_profile for now.
+      search_criteria['draw_profile'] = "medium"
+      wh_props = model_find_object(standards_data['water_heaters'], search_criteria, capacity = capacity_btu_per_hr, date = nil, area = nil, num_floors = nil, fan_motor_bhp = nil, volume = volumn_gal)
+      unless wh_props
+        OpenStudio.logFree(OpenStudio::Warn, 'openstudio.standards.WaterHeaterMixed', "For #{water_heater_mixed.name}, cannot find water heater properties, cannot apply efficiency standard.")
+        return false
+      end
+      uniform_energy_factor_base = wh_props['uniform_energy_factor_base']
+      uniform_energy_factor_volume_allowance = wh_props['uniform_energy_factor_volume_allowance']
```

Comment:
I do not see these two data in the `water_heaters` dataset. Could you verify and let me know which data file I should look into?

Reply:
The dataset is at \lib\openstudio-standards\standards\ashrae_90_1_prm\ashrae_90_1_prm_2019\data\ashrae_90_1_prm_2019.water_heaters.json.